torch/_dynamo/variables/tensor.py

```diff
@@ -336,7 +336,7 @@ def create(cls, tx, proxy, example_value=None, **options):
             return UserDefinedObjectVariable(example_value)
         elif isinstance(example_value, (torch.SymInt, torch.SymFloat)):
             proxy.node.meta["example_value"] = example_value
-            return cls(proxy, **options)
+            return DynamicShapeVariable(proxy, example_value, **options)
```

Comment:
The bug I saw was that we were incorrectly trying to call `torch.clamp()` on a symint (codepath [here](https://github.com/pytorch/pytorch/blob/0131a66ab6c8454c9ac7517641f63095b090e8cb/torch/_dynamo/variables/builtin.py#L357)), because we had a variable tracker advertising as a `TensorVariable`, who's inner example value (`proxy.node.meta["example_value"]`) was actually a SymInt.